ios/ImagePickerManager.m

```diff
@@ -269,33 +299,127 @@ - (void)checkPermission:(void(^)(BOOL granted)) callback {
         return info[UIImagePickerControllerImageURL];
-    } else {
+    }
+    else {
         return info[UIImagePickerControllerReferenceURL];
     }
 }
 
 @end
 
+@implementation ImagePickerManager(PHPickerViewControllerDelegate)
+
+- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14))
+{
+    if (results.count == 0) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [picker dismissViewControllerAnimated:YES completion:^{
+                self.callback(@[@{@"didCancel": @YES}]);
+            }];
+        });
+        return;
+    }
+
+    for (PHPickerResult *result in results) {
+        NSItemProvider *provider = result.itemProvider;
+
+        if ([provider canLoadObjectOfClass:[UIImage class]]) {
+            NSString *typeIdentifier = [provider.registeredTypeIdentifiers firstObject];
+            [provider loadObjectOfClass:[UIImage class]
+                      completionHandler:^(__kindof id<NSItemProviderReading> _Nullable object, NSError * _Nullable error) {
+                UIImage *image = object;
+                NSDictionary *response = [self makeResponseFromImage:image typeIdentifier:typeIdentifier];
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [picker dismissViewControllerAnimated:YES completion:^{
+                        self.callback(@[response]);
+                    }];
+                });
+            }];
+        }
+        else if ([provider hasItemConformingToTypeIdentifier:@"public.movie"]) {
+            [provider loadFileRepresentationForTypeIdentifier:@"public.movie"
+                                            completionHandler:^(NSURL * _Nullable url, NSError * _Nullable error) {
+                NSDictionary *response = [self makeResponseFromURL:url];
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [picker dismissViewControllerAnimated:YES completion:^{
+                        self.callback(@[response]);
+                    }];
+                });
+            }];
+        }
+    }
+}
+
+- (NSDictionary *)makeResponseFromImage:(UIImage *)image typeIdentifier:(NSString *)typeIdentifier {
+    NSString *fileType = [self fileTypeFromTypeIdentifier:typeIdentifier];
+
+    NSData *data;
+    if ([fileType isEqualToString:@"jpg"]) {
+        data = UIImageJPEGRepresentation(image, [self.options[@"quality"] floatValue]);
+    }
+    else {
+        data = UIImagePNGRepresentation(image);
+    }
+
+    if (![fileType isEqualToString:@"gif"]) {
+        image = [ImagePickerUtils resizeImage:image
+                                     maxWidth:[self.options[@"maxWidth"] floatValue]
+                                    maxHeight:[self.options[@"maxHeight"] floatValue]];
+    }
+
+    return [self makeResponseFromImage:image fileType:fileType data:data];
+}
+
+- (NSString *)fileTypeFromTypeIdentifier:(NSString *)typeIdentifier
```

Comment:
The `typeIdentifier` returns values like `com.compuserve.gif` or `public.png` so doing a bit of fuzzy matching here. There is [UTType](https://developer.apple.com/documentation/uniformtypeidentifiers/uttype) but that's for Swift only